Userland/Applications/PDFViewer/PDFViewer.h

```diff
@@ -29,7 +29,12 @@ class PDFViewer : public GUI::AbstractScrollableWidget {
     RefPtr<Gfx::Bitmap> get_rendered_page(u32 index);
     RefPtr<Gfx::Bitmap> render_page(const PDF::Page&);
 
+    void zoom_in();
+    void zoom_out();
+
     RefPtr<PDF::Document> m_document;
     u32 m_current_page_index { 0 };
-    HashMap<u32, RefPtr<Gfx::Bitmap>> m_rendered_pages;
+    Vector<HashMap<u32, RefPtr<Gfx::Bitmap>>> m_rendered_page_list;
```

Comment:
Sounds like we could use a way to allocate a `Gfx::Bitmap` on the heap, or multiples inside a large `mmap`'ed area

Reply:
> timer util

Yes, `start_timer(ms)` will end up calling `timer_event(Core::TimerEvent&)`

Reply:
Alright, just pushed that timer idea. Funny enough, the cache wasn't even working at all because I was taking the `HashMap` out of the vector by value... now zooming is like 5x faster lol.